packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -392,40 +392,54 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
     private async summarizeInternal(fullTree: boolean, trackState: boolean): Promise<ISummarizeInternalResult> {
         await this.realize();
 
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        const summarizeResult = await this.channel!.summarize(fullTree, trackState);
-        let pathPartsForChildren: string[] | undefined;
+        // Create attributes to be added to the summary.
+        const { pkg, isRootDataStore } = await this.getInitialSnapshotDetails();
 
-        if (!this.disableIsolatedChannels) {
-            // Wrap dds summaries in .channels subtree.
-            wrapSummaryInChannelsTree(summarizeResult);
-            pathPartsForChildren = [channelsTreeName];
+        let summarizeResult;
+        try {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            summarizeResult = await this.channel!.summarize(fullTree, trackState);
         }
+        catch (error) {
+            // Log helpful information (currently the package path) if there is a summary failure.
+            this.logger.sendErrorEvent(
+                { eventName: "failToSummarize", fluidDataStoreId: this.id, packagePath: JSON.stringify(pkg) },
+                error);
+        }
+        finally {
+            let pathPartsForChildren: string[] | undefined;
```

Comment:
Shouldn't this whole block be in try? What happens if any of the steps here fail? I don't think we need a finally block.

Reply:
Main point of this PR from what I understand is just to catch the summarize() failure, so that's why its targeted it like that.

Reply:
This whole code is part of summarize, so if anything fails, we end up not summarizing :)

Reply:
I do not think we can catch error - that would make summary succeed with empty snapshot for this data store.
I think we should log errors, but let them propagate same was as before, no?

Reply:
Yes, we need to throw after logging so it can propagate to upper layers.

Reply:
I think the idea was just to log in a `finally` block, no `catch` at all.

Reply:
Ah I see - my bad

Reply:
I don't think we want to always log it every time a data store is summarized as that will be very noisy and we don't really need that information. It should be done only when summarizing a data store fails.

Reply:
Oh, interesting point.  Never mind then

Reply:
I think if we want to put the whole code in try/catch, then we can put summarizeInternal in try/catch where it is getting called instead of having huge try/catch block.